trajectory_executor/src/trajectory_executor/trajectory_executor.cpp

```diff
@@ -0,0 +1,188 @@
+/*
+ * Copyright (C) 2018-2019 LEIDOS.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+#include "trajectory_executor/trajectory_executor.hpp"
+#include <ros/ros.h>
+#include <utility>
+#include <cav_msgs/SystemAlert.h>
+
+namespace trajectory_executor 
+{
+    cav_msgs::TrajectoryPlan trimFirstPoint(const cav_msgs::TrajectoryPlan plan) {
+        cav_msgs::TrajectoryPlan out(plan);
+        out.trajectory_points.erase(out.trajectory_points.begin());
+        return out;
+    }
+
+    TrajectoryExecutor::TrajectoryExecutor(int traj_frequency) :
```

Comment:
It does not look like this constructor will ever get called. Because it is likely that you will get traj_frequency from a ROS parameter, which is only from a ROS NodeHandle. But NodeHandle will generally created in the init() step.